cmd/kube-controller-manager/app/plugins.go

```diff
@@ -31,13 +31,25 @@ import (
 )
 
 // ProbeRecyclableVolumePlugins collects all persistent volume plugins into an easy to use list.
-func ProbeRecyclableVolumePlugins() []volume.VolumePlugin {
+func ProbeRecyclableVolumePlugins(config VolumeConfig) []volume.VolumePlugin {
 	allPlugins := []volume.VolumePlugin{}
 
-	// The list of plugins to probe is decided by the kubelet binary, not
+	// The list of plugins to probe is decided by this binary, not
 	// by dynamic linking or other "magic".  Plugins will be analyzed and
 	// initialized later.
-	allPlugins = append(allPlugins, host_path.ProbeVolumePlugins()...)
-	allPlugins = append(allPlugins, nfs.ProbeVolumePlugins()...)
+
+	// Each plugin can make use of VolumeConfig.  The single arg to this func contains *all* enumerated
+	// CLI flags meant to configure volume plugins.  From that single config, create an instance of volume.VolumeConfig
+	// for a specific plugin and pass that instance to the plugin's ProbeVolumePlugins(config) func.
+	hostPathConfig := volume.VolumeConfig{
+	// transfer attributes from VolumeConfig to this instance of volume.VolumeConfig
+	}
+	nfsConfig := volume.VolumeConfig{
+	// TODO transfer config.PersistentVolumeRecyclerTimeoutNFS and other flags to this instance of VolumeConfig
+	// Configuring recyclers will be done in a follow-up PR
+	}
```

Comment:
Because passing the whole struct would pass 1 big struct with specific values (PVRecycTimeoutForNFS) instead of a generic value (PVRecycTimeout).   

The design we settled on was to leave specifics in the binary (where the rubber has to meet the road and enumerated flags must be specific) and then use a more generic VolumeConfig which will never contain the word "NFS" or any other plugin detail.